src/Microsoft.TemplateEngine.Cli/TemplateListResolutionResult.cs

```diff
@@ -109,16 +110,46 @@ public bool TryGetAllInvokableTemplates(out IReadOnlyList<ITemplateMatchInfo> in
 
         public bool TryGetSingularInvokableMatch(out ITemplateMatchInfo template)
         {
-            IReadOnlyList<ITemplateMatchInfo> invokableMatches = _coreMatchedTemplates.Where(x => x.IsInvokableMatch() 
-                                                && (_hasUserInputLanguage
-                                                    || x.DispositionOfDefaults.Count == 0
-                                                    || x.DispositionOfDefaults.Any(y => y.Location == MatchLocation.DefaultLanguage && y.Kind == MatchKind.Exact))
-                                                    )
-                                                    .ToList();
-            
-            if (invokableMatches.Count == 1)
-            {
-                template = invokableMatches[0];
+            IReadOnlyList<ITemplateMatchInfo> invokableMatches = _coreMatchedTemplates.Where(x => x.IsInvokableMatch()).ToList();
+            IReadOnlyList<ITemplateMatchInfo> languageFilteredInvokableMatches;
+
+            if (_hasUserInputLanguage)
+            {
+                languageFilteredInvokableMatches = invokableMatches;
+            }
+            else
+            {
+                // check for templates with the default language
+                languageFilteredInvokableMatches = invokableMatches.Where(x => x.DispositionOfDefaults.Any(y => y.Location == MatchLocation.DefaultLanguage && y.Kind == MatchKind.Exact)).ToList();
+
+                if (languageFilteredInvokableMatches.Count == 0)
+                {
+                    // no templates with the default language.
+                    // Check if there are multiple languages among the results. If so, can't invoke.
+                    HashSet<string> languagesForMatches = new HashSet<string>();
+                    foreach (ITemplateMatchInfo templateInfo in invokableMatches)
+                    {
+                        if (templateInfo.Info.Tags.TryGetValue("language", out ICacheTag languageTag))
+                        {
+                            languagesForMatches.Add(languageTag.DefaultValue);
+                        }
+                    }
+
+                    if (languagesForMatches.Count > 1)
+                    {
+                        // there are multiple languages in the possible templates. No way to decide which one to use.
```

Comment:
This isn't strictly true is it? Couldn't precedence disambiguate (if one language of the template were preferred by the author over the others)?

Reply:
Up to now, we've only considered precedence among templates with the same group identity and language. I could add a check here so that if:
- all candidate templates are the same group identity
- user didn't specify a language
Then don't do the language check. These cases would fall through to the remaining existing checks, the last one being: take the highest precedence template if they're all the same group identity.

The check would change the new if at line 116 to be:
`if (_hasUserInputLanguage || _TemplateListResolver..AreAllTemplatesSameGroupIdentity(_coreMatchedTemplates))`